tools/throughput/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/cometbft/cometbft/types"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cometbft/cometbft/rpc/client/http"
+)
+
+type BlockMetrics struct {
+	mutex       sync.Mutex
```

Comment:
definitely not blocking due to usage, but needing a mutex for a stats msg seems offs

Reply:
good idea: f505269e67d6562b460d17f6244a3e88566c022f